tests/test_detectors.py

```diff
@@ -0,0 +1,143 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
+
+import unittest
+import glob
+import os
+import copy
+import torch
+from maskrcnn_benchmark.modeling.detector import build_detection_model
+from maskrcnn_benchmark.structures.image_list import to_image_list
+import utils
+
+
+CONFIG_FILES = [
+    # bbox
+    "e2e_faster_rcnn_R_50_C4_1x.yaml",
+    "e2e_faster_rcnn_R_50_FPN_1x.yaml",
+    "e2e_faster_rcnn_fbnet.yaml",
+
+    # mask
+    "e2e_mask_rcnn_R_50_C4_1x.yaml",
+    "e2e_mask_rcnn_R_50_FPN_1x.yaml",
+    "e2e_mask_rcnn_fbnet.yaml",
+
+    # keypoints
+    # TODO: fail to run for random model due to empty head input
+    # "e2e_keypoint_rcnn_R_50_FPN_1x.yaml",
+
+    # gn
+    "gn_baselines/e2e_faster_rcnn_R_50_FPN_1x_gn.yaml",
+    # TODO: fail to run for random model due to empty head input
+    # "gn_baselines/e2e_mask_rcnn_R_50_FPN_Xconv1fc_1x_gn.yaml",
```

Comment:
Could you expand on why those two cases fail?

Reply:
It seemed to fail when no input to the keypoint head. The error message is follow:
```
  File "/Users/stzpz/code/newstzpz/forked/maskrcnn-benchmark/maskrcnn_benchmark/modeling/roi_heads/roi_heads.py", line 53, in forward
    x, detections, loss_keypoint = self.keypoint(keypoint_features, detections, targets)
  File "/anaconda3/envs/maskrcnn_benchmark/lib/python3.7/site-packages/torch/nn/modules/module.py", line 491, in __call__
    result = self.forward(*input, **kwargs)
  File "/Users/stzpz/code/newstzpz/forked/maskrcnn-benchmark/maskrcnn_benchmark/modeling/roi_heads/keypoint_head/keypoint_head.py", line 42, in forward
    result = self.post_processor(kp_logits, proposals)
  File "/anaconda3/envs/maskrcnn_benchmark/lib/python3.7/site-packages/torch/nn/modules/module.py", line 491, in __call__
    result = self.forward(*input, **kwargs)
  File "/Users/stzpz/code/newstzpz/forked/maskrcnn-benchmark/maskrcnn_benchmark/modeling/roi_heads/keypoint_head/inference.py", line 15, in forward
    mask_prob, scores = self.keypointer(x, boxes)
  File "/Users/stzpz/code/newstzpz/forked/maskrcnn-benchmark/maskrcnn_benchmark/modeling/roi_heads/keypoint_head/inference.py", line 114, in __call__
    assert len(boxes) == 1
AssertionError
```

Reply:
Ok, thanks!
I'll see if I fix this in the future

Reply:
> It seemed to fail when no input to the keypoint head. The error message is follow:
> 
> ```
>   File "/Users/stzpz/code/newstzpz/forked/maskrcnn-benchmark/maskrcnn_benchmark/modeling/roi_heads/roi_heads.py", line 53, in forward
>     x, detections, loss_keypoint = self.keypoint(keypoint_features, detections, targets)
>   File "/anaconda3/envs/maskrcnn_benchmark/lib/python3.7/site-packages/torch/nn/modules/module.py", line 491, in __call__
>     result = self.forward(*input, **kwargs)
>   File "/Users/stzpz/code/newstzpz/forked/maskrcnn-benchmark/maskrcnn_benchmark/modeling/roi_heads/keypoint_head/keypoint_head.py", line 42, in forward
>     result = self.post_processor(kp_logits, proposals)
>   File "/anaconda3/envs/maskrcnn_benchmark/lib/python3.7/site-packages/torch/nn/modules/module.py", line 491, in __call__
>     result = self.forward(*input, **kwargs)
>   File "/Users/stzpz/code/newstzpz/forked/maskrcnn-benchmark/maskrcnn_benchmark/modeling/roi_heads/keypoint_head/inference.py", line 15, in forward
>     mask_prob, scores = self.keypointer(x, boxes)
>   File "/Users/stzpz/code/newstzpz/forked/maskrcnn-benchmark/maskrcnn_benchmark/modeling/roi_heads/keypoint_head/inference.py", line 114, in __call__
>     assert len(boxes) == 1
> AssertionError
> ```
Can you tell me how to solve this problem? We look forward to your reply.thanks a lot